lib/DriverTool/swift_api_digester_main.cpp

```diff
@@ -273,11 +273,10 @@ class RemovedAddedNodeMatcher : public NodeMatcher, public MatchedNodeListener {
   }
 
   static bool isNameTooSimple(StringRef N) {
-    static std::vector<std::string> SimpleNames = {"unit", "data", "log", "coding",
+    static llvm::StringSet<> SimpleNames = {"unit", "data", "log", "coding",
       "url", "name", "date", "datecomponents", "notification", "urlrequest",
       "personnamecomponents", "measurement", "dateinterval", "indexset"};
-    return std::find(SimpleNames.begin(), SimpleNames.end(), N) !=
-      SimpleNames.end();
+    return SimpleNames.contains(N);
```

Comment:
Do you have benchmarks to show that this is a demonstrable win? Often, linear search across small, fixed datasets like the above can be much faster than the memops required to execute a hash chain walk.

Reply:
Wow good to learn! Let me do a quick benchmark and get back to you (probably later this week!)

Reply:
Sorry for being late here. I made a quick benchmark project to compare the actual performance. The code is as simple as bellow. Please let me know if there is anything wrong, but seems like the change in this PR makes it more than 4x faster.

<img width="384" alt="Screen Shot 2022-06-24 at 1 13 47 PM" src="https://user-images.githubusercontent.com/2619092/175660971-5f3c9ba7-61c4-491e-866f-77c08c68105b.png">

```cpp
#include <llvm/ADT/StringSet.h>
#include <llvm/ADT/StringRef.h>
#include <benchmark/benchmark.h>

static bool isNameTooSimpleOriginal(llvm::StringRef N) {
    static std::vector<std::string> SimpleNames = {"unit", "data", "log", "coding",
        "url", "name", "date", "datecomponents", "notification", "urlrequest",
        "personnamecomponents", "measurement", "dateinterval", "indexset"};
    return std::find(SimpleNames.begin(), SimpleNames.end(), N) !=
      SimpleNames.end();
}

static bool isNameTooSimpleNew(llvm::StringRef N) {
    static llvm::StringSet<> SimpleNames = {"unit", "data", "log", "coding",
        "url", "name", "date", "datecomponents", "notification", "urlrequest",
        "personnamecomponents", "measurement", "dateinterval", "indexset"};
    return SimpleNames.contains(N);
}

static void Original(benchmark::State& state) {
  for (auto _ : state) {
    isNameTooSimpleOriginal("datecomponents");
    isNameTooSimpleOriginal("dummy");
  }
}

BENCHMARK(Original);

static void New(benchmark::State& state) {
  for (auto _ : state) {
    isNameTooSimpleNew("dummy");
    isNameTooSimpleNew("datecomponents");
  }
}

BENCHMARK(New);
```

Reply:
Oh wait.. If I do a release build, I got a worse result..
<img width="381" alt="Screen Shot 2022-06-24 at 2 39 34 PM" src="https://user-images.githubusercontent.com/2619092/175696502-c44893d8-05b6-46ef-9d2c-f5ffce4bd710.png">